src/index.js

```diff
@@ -0,0 +1,89 @@
+import { InstanceBase, runEntrypoint } from '@companion-module/base'
+import { initActions } from './actions.js'
+import { API } from './api.js'
+import { initFeedbacks } from './feedbacks.js'
+import { initProduct } from './models.js'
+import { setPolling } from './polling.js'
+import { initPresets } from './presets.js'
+import UpgradeScripts from './upgrades.js'
+import { initVariables, checkVariables } from './vars.js'
+import { getConfigDefinitions } from './config.js'
+
+class PTZControlerInstance extends InstanceBase {
+	constructor(internal) {
+		super(internal)
+		this.data = {}
+		this.polling = {}
+		this.product
+	}
+
+	async init(config) {
+		this.data = {
+			camera: 'NaN',
+			group: 'NaN',
+			port: 'NaN',
+		}
+
+		this.config = config
+		this.config.model = this.config.model || 'AW-RP50'
+		this.config.enablePolling = this.config.enablePolling || true
+
+		this.updateStatus('connecting')
+		this.api = new API(config)
+		this.configUpdated(config)
+		this.updateStatus('ok')
+	}
+
+	async destroy() {
+		if (this.socket !== undefined) {
+			this.socket.destroy()
+		}
+
+		if (this.polling.interval) {
+			clearInterval(this.polling.interval)
+		}
+
+		this.debug('destroy', this.id)
+	}
+
+	async configUpdated(config) {
+		const self = this
+
+		self.config = config
+		this.updateStatus('connecting')
+		this.updateStatus('unknown_error')
```

Comment:
Why? Status should only be updated if there really is some change of the status.

Reply:
The status update is [here](https://github.com/bitfocus/companion-module-panasonic-camera-controller/commit/0dd2d6815101917d957cd4ccc156c8105b97748a) from the beginning.
As a config update can disconnect the module and the controller, I suppose the original idea was to inform of this disconnection (and then, the "ok" status indicate that everything is back to normal).
I'll delete the "unknown error" one, which has less sense and which I forgot to remove.

Reply:
Lasts commits changed the way status is updated